### ISSUE TITLE
CI: deploy WASM version for every PR

### ISSUE THREAD

**Issue by sharkdp** [sharkdp/numbat#360]
It would be cool to be able to deploy the web version of Numbat in an automated way (to GitHub pages). This way, we could easily (1) test proposed features and bugfixes and (2) have a better test for the web version compared to the few headless-Firefox tests.

**Comment by bsidhom** [comment]
This is a great idea. I'm not sure how GH hooks work or what the scope of this is, but if we just wanted to auto-push the `master` branch to GH pages, that should be doable. On the other hand, it will not work in its current state unless the ECB proxy is updated to set the `Access-Control-Allow-Origin` header (either to `*` or the target GitHub Pages origin) or we switch to another proxy mechanism (e.g., using a [CloudFlare worker proxy](https://developers.cloudflare.com/workers/examples/cors-header-proxy/)).

**Comment by sharkdp** [comment]
> This is a great idea. I'm not sure how GH hooks work or what the scope of this is, but if we just wanted to auto-push the `master` branch to GH pages, that should be doable

Yes. With something like https://github.com/actions/deploy-pages. The problem seems to be with multiple deployments. I guess we would have to manually manage subfolders for branches, which seems ... not so clean. And would require cleanup jobs etc.

> On the other hand, it will not work in its current state unless the ECB proxy is updated to set the `Access-Control-Allow-Origin` header

I can do that whenever we need it.
